### PR TITLE
Update convolve avx2 functions

### PR DESCRIPTION
**Pull request by lzhangnj**
The total speed comparison of eb_av1_convolve_y_sr_avx2():

```
Blk Size | Tap 2    4    6    8
    4x 4 |   5.43  4.66 N/A  N/A
    4x 8 |   5.89  N/A  4.29 3.86
    8x 4 |   4.92  4.44 N/A  N/A
    8x 8 |   5.40  N/A  4.31 3.66
   8x 16 |   5.94  N/A  4.76 4.20
   16x 8 |   4.70  N/A  3.33 3.11
  16x 16 |   5.08  N/A  3.73 3.39
  16x 32 |   5.38  N/A  4.09 3.61
  32x 16 |   7.63  N/A  4.71 3.85
  32x 32 |   7.99  N/A  4.88 4.06
  32x 64 |   9.00  N/A  5.04 4.05
  64x 32 |  10.34  N/A  5.33 4.25
  64x 64 |  10.82  N/A  5.27 4.22
  64x128 |  10.12  N/A  5.12 4.02
 128x 64 |  11.73  N/A  5.43 4.24
 128x128 |  10.85  N/A  4.57 3.84
   4x 16 |   6.67  N/A  5.04 4.56
   16x 4 |   4.37  4.03 N/A  N/A
   8x 32 |   6.83  N/A  5.23 4.66
   32x 8 |   6.43  N/A  4.34 3.74
  16x 64 |   5.39  N/A  4.32 3.72
  64x 16 |   9.78  N/A  5.27 4.26
```

The total speed comparison of eb_av1_convolve_x_sr_avx2():

```
Blk Size | Tap 2    4    6    8
  4x  4  |    1.54 1.45 N/A  N/A
  4x  8  |    1.92 1.66 N/A  N/A
  8x  4  |    1.44 N/A  1.27 1.14
  8x  8  |    1.62 N/A  1.30 1.26
  8x 16  |    1.69 N/A  1.36 1.24
 16x  8  |    2.52 N/A  1.66 1.54
 16x 16  |    2.82 N/A  1.75 1.51
 16x 32  |    3.31 N/A  1.88 1.65
 32x 16  |    3.98 N/A  1.79 1.56
 32x 32  |    4.27 N/A  1.82 1.57
 32x 64  |    4.63 N/A  1.85 1.62
 64x 32  |    4.71 N/A  1.83 1.62
 64x 64  |    4.71 N/A  1.88 1.62
 64x128  |    4.48 N/A  1.82 1.63
128x 64  |    4.83 N/A  1.84 1.55
128x128  |    4.11 N/A  1.81 1.55
  4x 16  |    1.97 1.74 N/A  N/A
 16x  4  |    2.24 N/A  1.54 1.41
  8x 32  |    1.94 N/A  1.47 1.36
 32x  8  |    3.27 N/A  1.67 1.51
 16x 64  |    3.22 N/A  1.87 1.65
 64x 16  |    4.45 N/A  1.81 1.58
```

The total speed comparison of eb_av1_convolve_2d_sr_avx2():

```
         |        Half-pel        |    Other offsets
Blk Size | Tap 2    4    6    8   | Tap 2    4    6    8 
    4x 4 |  2.51  1.93 N/A  N/A   |  2.29  1.88 N/A  N/A
    4x 8 |  2.51  N/A  N/A  N/A   |  2.03  N/A  N/A  N/A 
    8x 4 |  2.16  N/A  N/A  N/A   |  2.10  N/A  N/A  N/A 
    8x 8 |  2.11  N/A  1.34 1.20  |  1.87  N/A  1.31 1.16
   8x 16 |  2.22  N/A  1.44 1.22  |  1.86  N/A  1.34 1.15
   16x 8 |  2.94  N/A  1.51 1.16  |  2.41  N/A  1.34 1.13
  16x 16 |  3.05  N/A  1.51 1.15  |  2.45  N/A  1.32 1.14
  16x 32 |  3.28  N/A  1.55 1.16  |  2.47  N/A  1.27 1.17
  32x 16 |  4.54  N/A  1.80 1.27  |  3.20  N/A  1.59 1.27
  32x 32 |  4.85  N/A  1.81 1.27  |  3.30  N/A  1.65 1.31
  32x 64 |  4.86  N/A  1.77 1.25  |  3.30  N/A  1.61 1.31
  64x 32 |  5.44  N/A  1.83 1.30  |  3.77  N/A  1.64 1.33
  64x 64 |  5.21  N/A  1.79 1.18  |  3.55  N/A  1.61 1.27
  64x128 |  2.76  N/A  1.55 1.13  |  3.13  N/A  1.36 1.19
 128x 64 |  4.34  N/A  1.64 1.20  |  3.47  N/A  1.53 1.22
 128x128 |  3.49  N/A  1.56 1.12  |  3.41  N/A  1.47 1.17
   4x 16 |  2.78  N/A  N/A  N/A   |  2.22  N/A  N/A  N/A
   16x 4 |  3.09  N/A  N/A  N/A   |  2.79  N/A  N/A  N/A
   8x 32 |  2.31  N/A  1.46 1.25  |  1.89  N/A  1.36 1.18
   32x 8 |  4.59  N/A  1.78 1.33  |  3.47  N/A  1.64 1.26
  16x 64 |  3.41  N/A  1.62 1.19  |  2.53  N/A  1.33 1.19
  64x 16 |  5.63  N/A  1.88 1.31  |  3.92  N/A  1.68 1.34
```

Related to PR #262

### CODE REVIEW

**Comment by JackZhouVCD** [comment]
I am seeing a 0.17% speed gain averaged over M{1,4,8}, bitstreams match after applying the pr

File: Source/Lib/Common/ASM_AVX2/transpose_avx2.c [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+ * SPDX - License - Identifier: BSD - 2 - Clause - Patent
+ */
+
+// C instance for unit test.
+
+#include "transpose_avx2.h"
+
+void transpose_8bit_16x16_reg128bit_instance_avx2(const __m128i *const in,
+    __m128i *const out) {
+	transpose_8bit_16x16_reg128bit_avx2(in, out);
```
Comment (1480c1) [inline Source/Lib/Common/ASM_AVX2/transpose_avx2.c]:
```suggestion
        transpose_8bit_16x16_reg128bit_avx2(in, out);
```
tab

**Comment by JackZhouVCD** [comment]
> I am seeing a 0.17% speed gain averaged over M{1,4,8}, bitstreams match after applying the pr

Tested this PR again, I am seeing a 1.38% speed gain averaged over M{1,4,8} and bitstreams match after applying the pr

**Comment by lzhangnj** [comment]
> Tested this PR again, I am seeing a 1.38% speed gain averaged over M{1,4,8} and bitstreams match after applying the pr

Thanks for helping testing!

**Comment by kylophone** [comment]
Tests are passing. LGTM.

**Comment by hassount** [comment]
is this still a [WIP] ?

**Comment by lzhangnj** [comment]
It can be merged.

I can check in eb_av1_jnt_convolve_2d_avx2() code later.

**Comment by hassount** [comment]
@ttrigui @mafonsoN can we review this and move it forward ?

**Comment by mafonsoN** [comment]
@kylophone has already reviewed it. Should be good to go.

**Pull request merged**